tensorflow/compiler/xla/service/gpu/ir_emission_utils.cc

```diff
@@ -312,17 +312,55 @@ llvm::Value* EmitPrintf(absl::string_view fmt,
        arguments_ptr});
 }
 
+// Helper function to emit call to AMDGPU shfl_down function.
+llvm::Value* EmitAMDGPUShflDown(llvm::Value* value, llvm::Value* offset,
+                                llvm::IRBuilder<>* b) {
+  llvm::Module* module = b->GetInsertBlock()->getModule();
+  CHECK_EQ(value->getType()->getPrimitiveSizeInBits(), 32);
+  auto* i32_ty = b->getInt32Ty();
+  llvm::FunctionCallee shfl_fn = module->getOrInsertFunction(
+      llvm_ir::AsStringRef("__ockl_readuplane_i32"),
+      llvm::FunctionType::get(/*Result=*/i32_ty, {i32_ty, i32_ty},
+                              /*isVarArg=*/false));
+  // AMDGPU device function requires first argument as i32.
+  llvm::Value* result = b->CreateCall(shfl_fn, {b->CreateBitCast(value, i32_ty) , offset});
+  // AMDGPU device function always returns an i32 type.
+  return b->CreateBitCast(result,value->getType());
+}
+
+// Helper function to emit call to NVPTX shfl_down intrinsic.
+llvm::Value* EmitNVPTXShflDown(llvm::Value* all_warps_mask, llvm::Value* value,
```

Comment:
s/all_warps_mask/mask/.  It's not necessarily the all-warps mask, i.e. -1.  (If it *were* always -1 then it shouldn't be a parameter at all...)

Reply:
I have skipped the parameter since its always being hardcoded to -1 right now.